### ISSUE TITLE
EOC Case where the last selected EOC has been deselected

### ISSUE THREAD

**Issue by frinzekt** [uwasystemhealth/IndEAA#51]
Steps to reproduce:
1. Create a justification for an EOC (eg. select EOC 1.1)
2. Press Save
3. Edit that EOC justification (eg. EOC 1.1)
4. Uncheck the checkbox for EOC 1.1
5. Save

What will happen is that in `handleSave`, `matchIndex=-1` (for a reason that I am unsure of). This means that, it is inserting a "new" entry in the database.

Another weird behaviour that I've seen from the Redux log is that `EVALUATION_PATCH_PENDING` is removing an array element.

I am not exactly sure what the problem is.

tldr; Cannot deselect the last EOC item
